### PR TITLE
start cleaning up the different NSE headers

### PR DESCRIPTION
**Pull request by zingale**
this is part of the way towards letting the approaches coexist

### CODE REVIEW

File: nse_solver/nse_check.H [hunk 1]
```diff
@@ -0,0 +1,1091 @@
...
+#include <eos_type.H>
+#include <network.H>
+#include <burner.H>
+#include <extern_parameters.H>
+#include <cmath>
+#include <AMReX_Array.H>
+#include <actual_network.H>
+#include <eos_composition.H>
+#include <nse_index.H>
+
```
Comment (zhichen3) [inline nse_solver/nse_check.H:14]: probably need to include `nse_solver.H` as well.

File: nse_solver/nse_check.H [hunk 2]
```diff
@@ -0,0 +1,1091 @@
...
+
+template <typename T>
+bool in_nse(T& state){
+
+  // This function returns the boolean that tells whether we're in nse or not
+  // Note that it only works with pynucastro network for now.
+
+  // Get the nse state which is used to compare nse molar fractions.
+  const auto nse_state = get_nse_state(state);
```
Comment (zhichen3) [inline nse_solver/nse_check.H:971]: I just realized I made a mistake here, it should use `get_actual_nse_state(state)`. Can you fix that for me?

Comment (zingale) [reply nse_solver/nse_check.H]: fixed

File: nse_solver/nse_check.H (hunk 1 above)
Comment (zingale) [reply nse_solver/nse_check.H]: done

**Review by zhichen3 — APPROVED** [review]

**Pull request merged**